zigzag/zigzag.py

```diff
@@ -13,259 +13,292 @@
-# Functions
-# ======================================================================================================================
-def _load_input_file(file_path, pprint_on_fail=False):
-    """Read and validate the input file contents.
+    def __init__(self, junit_xml_file_path, qtest_api_token, qtest_project_id, qtest_test_cycle, pprint_on_fail=False):
+        """ Create a ZigZag facade class object. The ZigZag class uses the Facade pattern to call out to
+        subsystems and sub Facades.
 
-    Args:
-        file_path (str): A string representing a valid file path.
-        pprint_on_fail (bool): A flag for enabling debug pretty print on schema failure.
+       Args:
+           junit_xml_file_path (str): A file path to a XML element representing a JUnit style testsuite response.
+           qtest_api_token (str): Token to use for authorization to the qTest API.
+           qtest_project_id (int): The target qTest project for the test results.
+           qtest_test_cycle (str): The parent qTest test cycle for test results. (e.g. Product Release codename "Queens")
+           pprint_on_fail (bool): A flag for enabling debug pretty print on schema failure.
+       """  # noqa
 
-    Returns:
-        ElementTree: An ET object already pointed at the root "testsuite" element.
+        swagger_client.configuration.api_key['Authorization'] = qtest_api_token
+        self._field_api = swagger_client.FieldApi()
+        self._junit_xml_file_path = junit_xml_file_path
+        self._qtest_project_id = qtest_project_id
+        self._qtest_test_cycle = qtest_test_cycle
+        self._pprint_on_fail = pprint_on_fail
+        self._failure_output_field_id = self._find_custom_field_id_by_label('Failure Output', 'test-runs')
 
-    Raises:
-        RuntimeError: invalid path.
-    """
+    def _find_custom_field_id_by_label(self, field_name, object_type):
+        """Find a custom field id by its label
 
-    root_element = 'testsuite'
-    junit_xsd = pytest_rpc.get_xsd()
+        Args:
+            field_name (str): The name of the custom field
+            object_type (str): The object type to search for the custom field on
 
-    try:
-        if os.path.getsize(file_path) > MAX_FILE_SIZE:
-            raise RuntimeError("Input file '{}' is larger than allowed max file size!".format(file_path))
+        Returns:
+            int: the id of the field that has the matching label
+        """
 
-        junit_xml_doc = etree.parse(file_path)
-    except (IOError, OSError):
-        raise RuntimeError("Invalid path '{}' for JUnitXML results file!".format(file_path))
-    except etree.ParseError:
-        raise RuntimeError("The file '{}' does not contain valid XML!".format(file_path))
+        for field in self._field_api.get_fields(self._qtest_project_id, object_type):
+            if field.label == field_name:
+                return field.id
```

Comment:
Why return when you can write the data directly onto the instance?

Reply:
Actually this method can look up many different fields on different kinds of objects.  I wouldn't know which value to write.